pkg/flowaggregator/flowaggregator.go

```diff
@@ -325,64 +325,76 @@ func (fa *flowAggregator) initExportingProcess() error {
 }
 
 func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
-	exportTicker := time.NewTicker(fa.exportInterval)
-	defer exportTicker.Stop()
 	go fa.collectingProcess.Start()
 	defer fa.collectingProcess.Stop()
 	go fa.aggregationProcess.Start()
 	defer fa.aggregationProcess.Stop()
+	go fa.flowRecordExpiryCheck(stopCh)
+
+	<-stopCh
+}
+
+func (fa *flowAggregator) flowRecordExpiryCheck(stopCh <-chan struct{}) {
+	expireTimer := time.NewTimer(fa.activeFlowRecordTimeout)
+
 	for {
 		select {
 		case <-stopCh:
 			if fa.exportingProcess != nil {
 				fa.exportingProcess.CloseConnToCollector()
 			}
+			expireTimer.Stop()
 			return
-		case <-exportTicker.C:
+		case <-expireTimer.C:
 			if fa.exportingProcess == nil {
 				err := fa.initExportingProcess()
 				if err != nil {
-					klog.Errorf("Error when initializing exporting process: %v, will retry in %s", err, fa.exportInterval)
-					// Initializing exporting process fails, will retry in next exportInterval
+					klog.Errorf("Error when initializing exporting process: %v, will retry in %s", err, fa.activeFlowRecordTimeout)
+					// Initializing exporting process fails, will retry in next cycle.
+					expireTimer = time.NewTimer(fa.activeFlowRecordTimeout)
```

Comment:
It can use `expireTimer.Reset` to avoid creating new timers.

Reply:
Was not aware of the method.. Reset makes sense here.